esp-hal/src/i2s.rs

```diff
@@ -847,7 +878,11 @@ mod private {
     }
 
     pub trait RegBlock:
-        crate::peripheral::Peripheral<P = Self> + PeripheralMarker + DmaEligible
+        crate::peripheral::Peripheral<P = Self>
+        + PeripheralMarker
+        + DmaEligible
+        + Into<super::AnyI2s>
+        + 'static
```

Comment:
:eyes: Why `'static`?

Reply:
So that we don't need bogus `+ 'd` lifetimes littered (thinking about `impl Peripheral<P = impl Into<Foo> + 'd> + 'd)`, though it kind of depends on what the constructors take and I'm not completely decided on that yet. Anyway, we don't have programmable hardware so our peirpheral instances don't go away.